--- a/lib/handyG/configure	2023-01-25 18:04:01.000000000 +0000
+++ b/lib/handyG/configure	2026-09-09 08:46:03.639323199 +0000
@@ -890,8 +890,8 @@
 		@\$(CC) \$(CXXFLAGS) -c \$<  -o \$@
 
 geval: libhandyg.a build/geval.o
-		@echo "LD \$@"
-		@\$(LD) `$STATIC && echo "--static" || true` -o \$@ build/geval.o -L. -lhandyg \$(LFLAGS)
+#		@echo "LD \$@"
+#		@\$(LD) `$STATIC && echo "--static" || true` -o \$@ build/geval.o -L. -lhandyg \$(LFLAGS)
 
 EOF
 
@@ -906,8 +906,8 @@
 		@\$(CC) \$(MCFLAGS) \$(CFLAGS) -o \$@ -c \$<
 
 handyG: build/mathlink.o libhandyg.a
-		@echo "LD \$@"
-		@\$(LD)  `$STATIC && echo "--static" || true` \$< libhandyg.a -o \$@ \$(LFLAGS) \$(MLFLAGS)
+#		@echo "LD \$@"
+#		@\$(LD)  `$STATIC && echo "--static" || true` \$< libhandyg.a -o \$@ \$(LFLAGS) \$(MLFLAGS)
 
 
 EOF
@@ -926,13 +926,12 @@
 build/test.o: checks/test.f90 \$(testdep)
 
 test: \$(objects) \$(testdep) build/test.o
-		@echo "LD \$@"
-		@\$(LD) -o \$@ \$^ \$(LFLAGS)
+#		@echo "LD \$@"
+#		@\$(LD) -o \$@ \$^ \$(LFLAGS)
 
-install: libhandyg.a handyg.mod ${CONF_MPREP:+handyG} geval
+install: libhandyg.a handyg.mod
 		mkdir -p \$(PREFIX)
 		mkdir -p \$(PREFIX)/bin
-		cp geval \$(PREFIX)/bin/geval
 		mkdir -p \$(PREFIX)/lib
 		cp libhandyg.a \$(PREFIX)/lib/
 		mkdir -p \$(PREFIX)/include
